packages/toolkit/src/query/core/buildMiddleware/invalidationByTags.ts

```diff
@@ -55,12 +71,27 @@ export const buildInvalidationByTagsHandler: InternalHandlerBuilder = ({
   }
 
   function invalidateTags(
-    tags: readonly FullTagDescription<string>[],
+    newTags: readonly FullTagDescription<string>[],
     mwApi: SubMiddlewareApi
   ) {
     const rootState = mwApi.getState()
     const state = rootState[reducerPath]
 
+    pendingTagInvalidations.push(...newTags)
+
+    if (state.config.invalidationBehavior === 'delayed') {
+      const hasPendingRequests = [
+        ...Object.values(state.queries),
+        ...Object.values(state.mutations),
+      ].some((x) => x?.status === QueryStatus.pending)
```

Comment:
❓ so the intent here is to delay invalidation until _every_ outstanding request is done, not just ones that might be overlapping?

Reply:
Hmm. I just wrote it up. It's like 20 lines more, but it introduces quite a bit of complexity and its a bit harder to optimize. I think I'd advocate for keeping it simple and looking if people have a problem with it. (I suspect they won't.)